forwards_paper.tex

```diff
@@ -822,46 +822,42 @@ \subsubsection*{Computational complexity}
 with simplify interval $s = 1$ so that redundant information is removed after every generation.
 Since every chromosome inherits material from both parents,
 without simplification this would produce tables of
-$2NT$ nodes and
-$4NT$ edges.
-Suppose that $T$ is large enough that all samples coalesce within the simulation with high probability,
-($T = 20N$, say).
+$NT$ nodes and
+$2NT$ edges.
+Suppose that $T$ is large enough that all samples coalesce within the simulation with high probability
+($T = 20N$, say, giving a total space requirement of $60N^2$).
 After simplification, we are left with the tree sequence describing the history
-of only the current generation of $2N$ individuals.
-If no coalescent events involve more than two lineages, this tree sequence has $4N-2$ edges to describe the leftmost tree;
+of only the current generation of $N$ individuals.
+If no coalescent events involve more than two lineages, this tree sequence has $2N-2$ edges to describe the leftmost tree;
 and each time the marginal tree changes along the sequence,
 four edges end and four new edges begin
 (except for changes affecting the root, which require fewer; see \citet{kelleher2016efficient}).
 Coalescent theory tells us that
-the expected total length of the edges of a marginal tree is approximately $4N\log(2N)$.
+the expected total length of the edges of a marginal tree is approximately $2N\log(N)$.
 Since we have taken one crossover per individual per generation (as is typical for a single chromosome),
 and we are measuring tree lengths in units of generations,
-the mean number of ancestral recombination events that occur somewhere on the marginal trees
-per unit of genetic length.
+this is the mean number of ancestral recombination events that occur somewhere on the marginal trees.
 Any such ancestral recombination might change the marginal gene tree;
 if it does, it will add four new edges.
 Not all such recombinations actually change the tree topology,
-so at most we expect $16N\log(2N)$ new edges after those required to describe the first tree.
+so at most we expect $8N\log(N)$ new edges after those required to describe the first tree.
 Similarly, not every new edge derives from a never-before-seen node,
 but the number of nodes is at most equal to the number of edges plus the sample size.
-This reduces the initial storage of $120 N^2$ items to
-% 4N - 2 + 2N + 4N log(2N) * 4 * 2 = 6N + 32 N log(2N)
-the space required after simplification of roughly
-\begin{align}
-    6N + 32 N \log(2N) .
-\end{align}
-At $N=10^4$, this reduces the space required to store the history of the entire population
-by a factor of 3,700.
+The space complexity after simplification is therefore $O(N + N\log N)$, since
+we need $O(N)$ space to store the first marginal tree, and $O(1)$ space
+for each of the $O(N \log N)$ trees that follow. Thus, we need $O(N^2)$ space
+to store the complete history of the simulation, but only $O(N + N \log N)$
```

Comment:
`O(N + N \log N) = O(N \log N)`.

Reply:
<scratches head> yes, I guess that's right.